app/controllers/chargeback_controller.rb

```diff
@@ -597,23 +702,48 @@ def cb_rate_set_form_vars
       # Add currencies to chargeback_controller.rb
-      @edit[:new][:details][i][:currency] = params[:currency] if params[:currency]
+      @edit[:new][:details][detail_index][:currency] = params[:currency] if params[:currency]
+
+      # Save tiers into @edit
+      (0..@sb[:num_tiers][detail_index].to_i - 1).each do |tier_index|
+        @edit [:new][:tiers][detail_index][tier_index] = {} unless @edit[:new][:tiers][detail_index][tier_index]
+        @edit[:new][:tiers][detail_index][tier_index][:fixed_rate] =
+          params["fixed_rate_#{detail_index}_#{tier_index}".to_sym] if params["fixed_rate_#{detail_index}_#{tier_index}".to_sym]
+        @edit[:new][:tiers][detail_index][tier_index][:variable_rate] =
+          params["variable_rate_#{detail_index}_#{tier_index}".to_sym] if params["variable_rate_#{detail_index}_#{tier_index}".to_sym]
+        @edit[:new][:tiers][detail_index][tier_index][:start] =
+          params["start_#{detail_index}_#{tier_index}".to_sym] if params["start_#{detail_index}_#{tier_index}".to_sym]
+        @edit[:new][:tiers][detail_index][tier_index][:end] =
+          params["end_#{detail_index}_#{tier_index}".to_sym] if params["end_#{detail_index}_#{tier_index}".to_sym]
+      end
     end
   end
 
   def cb_rate_set_record_vars
-    @edit[:new][:details].each_with_index do |_rate, i|
-      @sb[:rate_details][i].rate               = @edit[:new][:details][i][:rate]
-      @sb[:rate_details][i].per_time           = @edit[:new][:details][i][:per_time]
-      @sb[:rate_details][i].per_unit           = @edit[:new][:details][i][:per_unit]
+    @edit[:new][:details].each_with_index do |_detail, detail_index|
+      @sb[:rate_details][detail_index].per_time           = @edit[:new][:details][detail_index][:per_time]
+      @sb[:rate_details][detail_index].per_unit           = @edit[:new][:details][detail_index][:per_unit]
       # C: Record the currency selected in the edit view, in my chargeback_rate_details table
-      @sb[:rate_details][i].chargeback_rate_detail_currency_id = @edit[:new][:details][i][:currency]
-      @sb[:rate_details][i].chargeback_rate_id = @sb[:rate].id
+      @sb[:rate_details][detail_index].chargeback_rate_detail_currency_id = @edit[:new][:details][detail_index][:currency]
+      @sb[:rate_details][detail_index].chargeback_rate_id = @sb[:rate].id
+      # Save tiers into @sb
+      @edit[:new][:tiers][detail_index].each_with_index do |_tier, tier_index|
+        @sb[:tiers][detail_index][tier_index] =
+          ChargebackTier.new(:start                     => @edit[:new][:tiers][detail_index][tier_index][:start],
+                             :end                       => @edit[:new][:tiers][detail_index][tier_index][:end],
+                             :fixed_rate                => @edit[:new][:tiers][detail_index][tier_index][:fixed_rate],
+                             :variable_rate             => @edit[:new][:tiers][detail_index][tier_index][:variable_rate],
```

Comment:
`@edit[:new][:tiers][detail_index]` is already in a unused local variable `_tier`, rename that local variable and use it please.